.github/workflows/prepare-matrix.ps1

```diff
@@ -0,0 +1,55 @@
+
+param(
+    [Parameter(Mandatory=$false)]
+    [string]$MatrixFileName = "quic_matrix.json"
+)
+
+Set-StrictMode -Version "Latest"
+$PSDefaultParameterValues["*:ErrorAction"] = "Stop"
+
+
+$MatrixJson = Get-Content -Path .\.github\workflows\$MatrixFileName | ConvertFrom-Json
+
+
+$AzureJson = @()
+$LabJson = @()
+
+foreach ($entry in $MatrixJson) {
+    if ($entry.env -match "azure") {
+        $Windows2022Pool = "netperf-aztestpool"
+        $Ubuntu2004Pool = "netperf-aztestpool-linux"
+        $client = $entry.PSObject.Copy()
+        $server = $entry.PSObject.Copy()
+        $env_str = "$($entry.os)-$($entry.arch)-$($entry.tls)-$($entry.io)"
+        if ($entry.os -match "windows-2022") {
+            $client | Add-Member -MemberType NoteProperty -Name "assigned_pool" -Value $Windows2022Pool
+            $server | Add-Member -MemberType NoteProperty -Name "assigned_pool" -Value $Windows2022Pool
+            $client | Add-Member -MemberType NoteProperty -Name "role" -Value "client"
+            $server | Add-Member -MemberType NoteProperty -Name "role" -Value "server"
+            $client | Add-Member -MemberType NoteProperty -Name "env_str" -Value $env_str
+            $server | Add-Member -MemberType NoteProperty -Name "env_str" -Value $env_str
+            $client | Add-Member -MemberType NoteProperty -Name "remote_powershell_supported" -Value 'TRUE'
+            $server | Add-Member -MemberType NoteProperty -Name "remote_powershell_supported" -Value 'TRUE'
+            $AzureJson += $client
+            $AzureJson += $server
+        }
+        if ($entry.os -match "ubuntu-20.04") {
+            $client | Add-Member -MemberType NoteProperty -Name "assigned_pool" -Value $Ubuntu2004Pool
+            $server | Add-Member -MemberType NoteProperty -Name "assigned_pool" -Value $Ubuntu2004Pool
+            $client | Add-Member -MemberType NoteProperty -Name "role" -Value "client"
+            $server | Add-Member -MemberType NoteProperty -Name "role" -Value "server"
+            $client | Add-Member -MemberType NoteProperty -Name "env_str" -Value $env_str
+            $server | Add-Member -MemberType NoteProperty -Name "env_str" -Value $env_str
+            $client | Add-Member -MemberType NoteProperty -Name "remote_powershell_supported" -Value 'FALSE'
+            $server | Add-Member -MemberType NoteProperty -Name "remote_powershell_supported" -Value 'FALSE'
+            $AzureJson += $client
+            $AzureJson += $server
+        }
```

Comment:
Pull out the common commands and put them outside the `if` statements.

Reply:
Done.